### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits/VanKampen): Properties of van Kampen colimits

### PR DESCRIPTION
**Pull request by erdOne**
Also moves the definition of van Kampen colimits from `CategoryTheory/Extensive.lean` into a new file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/VanKampen.lean [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+/-!
+
+# Universal colimits and van Kampen colimits
+
+## Main definitions
+- `CategoryTheory.IsVanKampenColimit`: A (colimit) cocone over a diagram `F : J ⥤ C` is universal
+  if it is stabler under pullbacks.
+- `CategoryTheory.IsUniversalColimit`: A (colimit) cocone over a diagram `F : J ⥤ C` is van
+  Kampen if for every cocone `c'` over the pullback of the diagram `F' : J ⥤ C'`,
+  `c'` is colimiting iff `c'` is the pullback of `c`.
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/VanKampen.lean:24]: * The lean names and the docstrings seem to be swapped
* `stabler` should be `stable`
* Could you please add a reference to literature on Van Kampen colimits?

Comment (erdOne) [reply Mathlib/CategoryTheory/Limits/VanKampen.lean]: Done.

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**